### PR TITLE
feat(gltf): ext-structural-metadata encoding

### PR DESCRIPTION
**Pull request by mspivak-actionengine**

### CODE REVIEW

File: modules/gltf/src/lib/extensions/EXT_structural_metadata.ts [hunk 1]
```diff
@@ -701,3 +710,229 @@ function getEnumByValue(
...
+    extension = scenegraph.addExtension(EXT_STRUCTURAL_METADATA_NAME);
+  }
+
+  extension.schema = createSchema(propertyAttributes, classId, extension.schema);
+  const table = createPropertyTable(propertyAttributes, classId, extension.schema);
+  if (!extension.propertyTables) {
+    extension.propertyTables = [];
+  }
+
+  // scenegraph.createBinaryChunk();
```
Comment (belom88) [inline modules/gltf/src/lib/extensions/EXT_structural_metadata.ts:792]: clean up

File: modules/gltf/src/lib/extensions/EXT_structural_metadata.ts [hunk 2]
```diff
@@ -701,3 +710,229 @@ function getEnumByValue(
...
+      throw new Error('Illegal values in attributes');
+    }
+
+    const classProperty = schemaClass?.properties[attribute.name];
+    if (classProperty) {
+      // const tableProperty = createPropertyTableProperty(attribute, classProperty, scenegraph);
+      if (!table.properties) {
+        table.properties = {};
+      }
+      // table.properties[attribute.name] = tableProperty;
```
Comment (belom88) [inline modules/gltf/src/lib/extensions/EXT_structural_metadata.ts:850]: Clean up

File: modules/gltf/src/lib/extensions/EXT_structural_metadata.ts [hunk 3]
```diff
@@ -701,3 +710,229 @@ function getEnumByValue(
...
+  UINT16: Uint16Array,
+  INT32: Int32Array,
+  UINT32: Uint32Array,
+  INT64: Int32Array,
+  UINT64: Uint32Array,
+  FLOAT32: Float32Array,
+  FLOAT64: Float64Array
+};
+
+function propertyDataScalar(array: number[], componentType: string): TypedArray {
```
Comment (belom88) [inline modules/gltf/src/lib/extensions/EXT_structural_metadata.ts:894]:
```suggestion
function createPropertyDataScalar(array: number[], componentType: string): TypedArray {
```

File: modules/gltf/src/lib/extensions/EXT_structural_metadata.ts [hunk 4]
```diff
@@ -701,3 +710,229 @@ function getEnumByValue(
...
+    numberArray.push(Number(value));
+  }
+  const Construct = COMPONENT_TYPE_TO_ARRAY_CONSTRUCTOR[componentType];
+  if (!Construct) {
+    throw new Error('Illegal component type');
+  }
+  return new Construct(numberArray);
+}
+
+function propertyDataString(strings: string[]): {
```
Comment (belom88) [inline modules/gltf/src/lib/extensions/EXT_structural_metadata.ts:906]:
```suggestion
function createPropertyDataString(strings: string[]): {
```

**Review by belom88 — APPROVED** [review]

File: modules/gltf/src/lib/extensions/EXT_structural_metadata.ts (hunk 2 above)
Comment (mspivak-actionengine) [reply modules/gltf/src/lib/extensions/EXT_structural_metadata.ts:850]: Done.

File: modules/gltf/src/lib/extensions/EXT_structural_metadata.ts (hunk 3 above)
Comment (mspivak-actionengine) [reply modules/gltf/src/lib/extensions/EXT_structural_metadata.ts:894]: Done.

File: modules/gltf/src/lib/extensions/EXT_structural_metadata.ts (hunk 4 above)
Comment (mspivak-actionengine) [reply modules/gltf/src/lib/extensions/EXT_structural_metadata.ts:906]: Done.

**Pull request merged**